# 01 — Exploração (EDA)

**Objetivo:** visualizar e descobrir. Nada de modelagem aqui.

Roteiro:
1. Carregar `data/raw/survey.csv` (rodar antes: `python -m src.data`)
2. `df.info()`, `df.describe()` — tipos, nulos, tamanho
3. Distribuição do alvo `treatment` — desbalanceado?
4. Problemas de qualidade: `Age` absurdos, `Gender` texto livre (quantas variações?)
5. Distribuições das features vs alvo (countplots)
6. **Anotar descobertas em células markdown** — vira insumo pra `docs/technical.md`

In [8]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

sns.set_theme()
df = pd.read_csv("../data/raw/survey.csv")
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1259 entries, 0 to 1258
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype
---  ------                     --------------  -----
 0   Timestamp                  1259 non-null   str  
 1   Age                        1259 non-null   int64
 2   Gender                     1259 non-null   str  
 3   Country                    1259 non-null   str  
 4   state                      744 non-null    str  
 5   self_employed              1241 non-null   str  
 6   family_history             1259 non-null   str  
 7   treatment                  1259 non-null   str  
 8   work_interfere             995 non-null    str  
 9   no_employees               1259 non-null   str  
 10  remote_work                1259 non-null   str  
 11  tech_company               1259 non-null   str  
 12  benefits                   1259 non-null   str  
 13  care_options               1259 non-null   str  
 14  wellness_program           1259 non

In [9]:
# Distribuição do alvo
df["treatment"].value_counts(normalize=True)

treatment
Yes    0.505957
No     0.494043
Name: proportion, dtype: float64

## Descobertas

Alvo equilibrado: 50,6% yes / 49,4% no. Boa notícia, modelo não vicia numa classe. A suposição de desbalanceamento no README estava errada; F1 continua métrica válida. 

Problemas de normalização nos dados: Age com lixo extremo: mínimo -1726, máximo 99999999999. 8 linhas removidas pelo filtro 18–75. 

Gender: 49 variações de texto livre ("male", "M", "make", "maile", "cis male", "male-ish"...). Normalizado: 971 male / 242 female / 38 other. 

Nulos: work_interfere 264 (21% — muito pra descartar, viraram "Unknown"), self_employed 18, state 515, comments 1095. 

Sinais fortes no dado (taxa de quem busca tratamento): 

family_history: yes 74% vs no 35% — sinal mais forte
work_interfere: often 85% vs never 14% — quase separa sozinho
mulheres 69% vs homens 45%
remote_work e tech_company: ~50% dos dois lados — quase inúteis